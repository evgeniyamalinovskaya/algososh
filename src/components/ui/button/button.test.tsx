import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from "@testing-library/react";
import { Button } from "./button";

//Проверяем при помощи снэпшотов корректную отрисовку:
describe('Тестирование компонента Button', () => {
    it('Кнопка c текстом', () => {
        //Рендерим Button с текстом
        const tree = renderer.create(<Button text={'test text'} />).toJSON();
        //Проверка, toMatchSnapshot() сравнивает входные данные выходным данным
        expect(tree).toMatchSnapshot();
    });

    it('Кнопка без текста', () => {
        const tree = renderer.create(<Button />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Заблокированной кнопки', () => {
        const tree = renderer.create(<Button disabled />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Кнопка с индикацией загрузки', () => {
        const tree = renderer.create(<Button isLoader />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Нажатие на кнопку вызывает alert', () => {
        //Имитация браузерного alert
        window.alert = jest.fn();
        //Рендерим компонент
        render(
            <Button onClick={() => window.alert('text')}
                    text={'test text'}
            />)
        //Находим элемент кнопки
        const button = screen.getByText('test text');
        //Имитируем нажатие на кнопку
        fireEvent.click(button);
        //Проверяем, что alert сработал с правильным текстом предупреждения
        expect(window.alert).toHaveBeenCalledWith('text');
    })
});
