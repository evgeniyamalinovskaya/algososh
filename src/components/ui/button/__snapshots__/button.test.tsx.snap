// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование компонента Button Заблокированной кнопки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тестирование компонента Button Кнопка c текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    test text
  </p>
</button>
`;

exports[`Тестирование компонента Button Кнопка без текста 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тестирование компонента Button Кнопка с индикацией загрузки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;
